[Jonyxzx] iP

Jonyxzx opened the pull request:
# AcePro
> "Chatbots will be your new best friend." — Christine Crandell ([Forbes](https://www.forbes.com/sites/christinecrandell/2016/10/23/chatbots-will-be-your-new-best-friend/?sh=1a3bb519a245))
## AcePro is build for user who loves basic GUI to keep track of tasks.
### Reasons to use AcePro:
- Simple to use
- Clean GUI
- Hassle free
### Use AcePro now for free in 3 ~~hard~~ **simple** step 🤪:
1. Clone this repository
2. Open Intellij IDEA
3. Run the launcher
### Features:
- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
### If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Jonyxzx committed:
Level 0 added

Jonyxzx committed:
Level 1 added

Jonyxzx committed:
Level 2 added

Jonyxzx committed:
Level 3 + A-Classes added

Jonyxzx committed:
Level 4 + A-Inheritance added

Jonyxzx committed:
Level-5 + A-TextUiTesting added

Jonyxzx committed:
Level 6 + A-Exceptions + A-Collections

Jonyxzx committed:
Refactor the code to apply more OOP concepts and implement save data

Jonyxzx committed:
Add the ability for deadline and event to recognise time date format

Jonyxzx committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

Jonyxzx committed:
Merge gradle into the main branch and add JUnit testcases

Jonyxzx committed:
Create jar files and fix output display errors

Jonyxzx committed:
Add the find feature to search for keyword in the list

Jonyxzx committed:
Add javadoc for all public methods and classes

Jonyxzx committed:
Add checkstyle plugins and rewrite the code to fix the checkstyle errors

Jonyxzx committed:
Add GUI to the chatbot and transform the program into a GUI based application

Jonyxzx committed:
Refactor the code to fit the GUI support

Jonyxzx committed:
Add the ability to take in multiple inputs at one go

Jonyxzx committed:
Fix some wrapped text issue due to long text output

cmHuang777 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Implements classes define specific actions for each command.
+ */
+public interface Command {
```
I like how you split up the commands into different class! It is a lot of effort 👍

cmHuang777 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+public class Duke {
+    public final Ui ui;
+    private final Storage storage;
+    private TaskList tasks;
```
I like how you use the plural form to indicate the TaskList object contains many Tasks objects!

cmHuang777 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,56 @@
+        String keyword = parts[0];
+        String details = parts.length == 2 ? parts[1] : "";
+
+        switch (keyword) {
+        case "bye":
+            return new ByeCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            return new MarkCommand(details);
+        case "unmark":
+            return new UnmarkCommand(details);
+        case "todo":
+            return new TodoCommand(details);
+        case "deadline":
+            return new DeadlineCommand(details);
+        case "event":
+            return new EventCommand(details);
+        case "delete":
+            return new DeleteCommand(details);
+        case "find":
+            return new FindCommand(details);
+        default:
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
```
I like how you matched the switch case indentation according to the coding standard in cs2103!

cmHuang777 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+            Task curr = new Deadline(partDeadline[0], partDeadline[1].trim());
+            tasks.add(curr);
+            ui.sendMessage("Got it. I've added this task:\n" + "\t" + curr + "\n"
+                    + "Now you have " + tasks.size() + " tasks in the list.");
```
I like how you placed the + sign on the start of new line instead of at the end of the line, to match the cs2103 style !

cmHuang777 reviewed (commented):
Overall, I found your code easy to read, with good coding style. 👍 
Nothing I can nit pick on!

RiyaMehta2211 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
```
good job on using appropriate packages to categorize the classes!

RiyaMehta2211 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+            try {
+                Command c = Parser.parse(input.trim());
+                c.execute(tasks, ui, storage);
+            } catch (DukeException | DateTimeException e) {
```
It's good that you accounted for the DateTimeException as well which can result from invalid inputs for date and time

RiyaMehta2211 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * Adds a new Deadline task to the task list.
+ */
+public class DeadlineCommand implements Command {
```
I like how you implemented the Command interface, splitting different instructions into commands and executing them.

RiyaMehta2211 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,56 @@
+        String details = parts.length == 2 ? parts[1] : "";
+
+        switch (keyword) {
+        case "bye":
```
Dividing the instructions into different classes and then creating their instances in a switch case definitely makes the code very readable. Good job!

RiyaMehta2211 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,80 @@
+ * Manages reading from and writing to a specified file.
+ */
+public class Storage {
+    private final String filePath;
```
Good job setting the fields private or protected which is in line with improved code quality. Also making them final to ensure the field values don't get accidentally modified. Overall, the code design has been well thought out, especially in view of the extensions to be done in the future

RiyaMehta2211 reviewed (commented):
Overall, I found the code very readable. There was a lot of effort that went in as there were many classes being made and used appropriately. I found the code has an extensible design as the different subclasses of Command can be modified easily to accommodate more changes in future. The fields were also set to final to prevent accidental modifications in future.

Jonyxzx committed:
Fix minor naming issue by following the coding standard

Jonyxzx committed:
Add assertions for input validation and error handling

Assert statements are used to ensure that the object state adhere to expectations at all times.
By doing so, we can ensure that everything goes as intended during runtime, which is useful as
the chatbot does not adopt input/output style.

The assertions will catch any discrepencies and prevent any unexpected behaviour.

Jonyxzx committed:
Merge pull request #1 from Jonyxzx/branch-A-CodeQuality

Fix minor naming issue by following the coding standard

Jonyxzx committed:
Merge branch 'master' into branch-A-Assertions

Jonyxzx committed:
Merge pull request #2 from Jonyxzx/branch-A-Assertions

Add assertions for input validation and error handling

Jonyxzx committed:
Implement the find command by using java streams

Jonyxzx committed:
Add Undo function into the chatbot to undo previous command

There is instance where the user require to revert back the command, without thinking of using the counter command

The "undo" command can be use to revert back the valid command executed previously, by automatically execute the counter
command. The stack data structure is used to store all the valid commands and the undo can be repeatedly use to revert all the way
to the fresh state.

By doing so, it improve the convenience for the user to modify their tasks storing.

Jonyxzx committed:
Improve the GUI by adding extra elements on the interface

Jonyxzx committed:
Add more exception handling for invalid user inputs

Jonyxzx committed:
Add CI into the project

Jonyxzx committed:
Revert "Add CI into the project"

This reverts commit 532a508bbce534e4536ccf4093878f504ddbe3e7.

Jonyxzx committed:
Revert "Revert "Add CI into the project""

This reverts commit 4e10d4b1c6978d3e171f278c8a51ea5b51158e44.

Jonyxzx committed:
Revert "Revert "Revert "Add CI into the project"""

This reverts commit 75edd05589b4b95eb2ee5220c35430a03f1abbcc.

Jonyxzx committed:
Update README.md

Jonyxzx committed:
Fix the image issue as it does not load as expected

Jonyxzx committed:
Update README.md

Jonyxzx committed:
Delete command bug fixes and file creation bug fixes

Jonyxzx committed:
Merge branches 'master' and 'master' of https://github.com/Jonyxzx/ip

Jonyxzx committed:
Update README.md

Jonyxzx committed:
Fix the file reading bug, find command bug, mark unmark bug

Jonyxzx committed:
Update README.md

Jonyxzx committed:
Minor string bug
